test/test_diff.py

```diff
@@ -177,11 +179,25 @@ def test_diff_asdf(capsys, jwst_shared_cache_state, jwst_data):
 
     output = capsys.readouterr().out
 
-    assert output == """        ndarrays differ by contents
-        ndarrays differ by contents
```

Comment:
The output of `asdftool diff` changed in asdf 3 and no longer matches this expected output. https://github.com/asdf-format/asdf/pull/1672 fixes the issue that caused this test to raise an exception and improved the output to better match the other tree differences.